File: 01_week2/lcm/lcm.py
# Uses python3
import sys

def gcd_fast(a, b):
	current_gcd = a
	div = b
	if a==0:
		return b
	if b==0:
		return a
	rem = div % current_gcd
	while rem!=0:
		div = current_gcd
		current_gcd = rem
		rem = div%current_gcd
		
	return current_gcd

def lcm_naive(a, b):
    for l in range(1, a*b + 1):
        if l % a == 0 and l % b == 0:
            return l

    return a*b

def lcm_fast(a, b):
	return (a * b)//gcd_fast(a, b)

if __name__ == '__main__':
    #input = sys.stdin.read()
    a, b = map(int, input().split())
    print(lcm_fast(a, b))

File: 01_week2/lcm/lcm.cpp
#include <iostream>

int gcd(int a, int b){
	int current_gcd=a;
	int rem,div;
	div=b;
	if(a==0)
		return b;
	if(b==0)
		return a;
	rem=div%current_gcd;
	while(rem!=0){
		div= current_gcd;
		current_gcd=rem;
		rem=div%current_gcd;
	}
	return current_gcd;
}

long long lcm_naive(int a, int b) {
  for (long l = 1; l <= (long long) a * b; ++l)
    if (l % a == 0 && l % b == 0)
      return l;

  return (long long) a * b;
}

long long lcm_fast(int a, int b) {

  return ((long long) a * b)/gcd(a, b);
}

int main() {
  int a, b;
  std::cin >> a >> b;
  std::cout << lcm_fast(a, b) << std::endl;
  return 0;
}
